rero_ils/modules/items/api/circulation.py

```diff
@@ -215,31 +218,57 @@ def is_action_validate_request_possible(self, **kwargs):
             raise NoCirculationAction(
                 'No circulation action is possible')
 
-    def prior_checkin_actions(self, item, **kwargs):
+    def prior_extend_loan_actions(self, **kwargs):
+        """Actions to execute before an extend_loan action."""
+        loan_pid = kwargs.get('pid')
+        if not loan_pid:
+            loan = self.get_first_loan_by_state(LoanState.ITEM_ON_LOAN)
+        else:
+            loan = Loan.get_record_by_pid(loan_pid)
+
+        # Check extend is allowed
+        checked_out = True  # we consider loan as checked-out
+        have_request = False
+        if not loan_pid and not loan:
+            # no loand_pid, so we check first_loan_by_state, without result:
+            # item was not checked out
+            checked_out = False
+        if LoanState.PENDING in self.get_loan_states_for_an_item():
```

Comment:
`elif` ?

Reply:
Not needed anymore as I moved previous `if` in another block.